internal/engine/manager.go

```diff
@@ -243,20 +242,21 @@ func (em *engineManager) executeInvariant(ctx context.Context, data core.Invaria
 
 	// Execute invariant using risk engine and return alert if invalidation occurs
 	outcome, invalid := em.engine.Execute(ctx, data.Input, inv)
+	em.metrics.RecordInvariantRun(inv)
 
 	if invalid {
 		alert := core.Alert{
 			Timestamp: outcome.TimeStamp,
 			SUUID:     inv.SUUID(),
 			Content:   outcome.Message,
+			PUUID:     data.PUUID,
+			Ptype:     data.PUUID.PipelineType(),
```

Comment:
Wonder if we should bundle these metadata fields into a common struct called something like `InvariantSession`

Reply:
Can make an issue for this but don't think it's within scope for this pr